src/Calibur.sol

```diff
@@ -137,6 +137,12 @@ contract Calibur is
                 // Forces the compiler to optimize for smaller bytecode size.
                 if (uint256(digest) == ~wrappedSignature.length / 0xffff * 0x7739) return 0x77390001;
             }
+            // If the signature is 65 bytes, it is a raw ECDSA signature and MUST be verified as the root key
+            else if (wrappedSignature.length == 65) {
+                if (KeyLib.toRootKey().verify(digest, wrappedSignature)) {
```

Comment:
if the signature is 65 bytes (or 64 re comment above) and it DOES not pass the verification against the root key I think we should early return false no?

Reply:
And I think we cannot early return when it is valid as we need to still process the hook call in the valid case

Reply:
Can you also leave a comment about why we can skip the hook callsite when the signature is valid against the root key? I got confused and then remembered we don't support hooks essentially on the root key path

Reply:
I did not return false if it failed here because we had erc1271safeCaller, but now that we removed it we should return false here. Good catch